web_src/js/features/user-auth-webauthn.js

```diff
@@ -113,10 +113,10 @@ function webauthnRegistered(newCredential) {
 
 function webAuthnError(errorType, message) {
   $('#webauthn-error [data-webauthn-error-msg]').hide();
-  if (errorType === 0 && message && message.length > 1) {
-    $(`#webauthn-error [data-webauthn-error-msg=0]`).text(message);
-    $(`#webauthn-error [data-webauthn-error-msg=0]`).show();
-  } else {
+  if (errorType === 'general') {
+    $(`#webauthn-error [data-webauthn-error-msg=general]`).text(message || 'unknown error');
+    $(`#webauthn-error [data-webauthn-error-msg=general]`).show();
+  } else if (errorType !== 'general') {
     $(`#webauthn-error [data-webauthn-error-msg=${errorType}]`).show();
   }
```

Comment:
```suggestion
  const $errorGeneral = $(`#webauthn-error [data-webauthn-error-msg=general]`);
  if (errorType === 'general') {
    $errorGeneral.show().text(message || 'unknown error');
  } else {
    const $errorTyped = $(`#webauthn-error [data-webauthn-error-msg=${errorType}]`);
    if ($errorTyped.length) {
      $errorTyped.show();
    } else {
      $errorGeneral.show().text(`unknown error type: ${errorType}`);
    }
  }
```

Make sure in any case (even if a developer made something wrong to the error type), users could always see a related error message. Anyway, we do not have unit test for this ....